### PR TITLE
fix: Installer improvements and bug fixes

### LINKED ISSUE #175: Introduce better logging for cron sshnpd mechanism
### Describe the bug

Line in question: https://github.com/atsign-foundation/sshnoports/blob/5ee87985319f5f3b1ed12d1d8bc3234f58ef0ff7/scripts/install_sshnpd#L215

Logs a potentially infinite size file into a hidden directory.

### LINKED ISSUE #203: FileSystemException inside ~/.sshnp
### Describe the bug

If you've previously installed sshnpd (have not yet tested for sshnp), there's some weird FileSystemException.

Doing `rm -rf ~/.sshnp` then re-running the script/binary (e.g. `./sshnpd@server_0`) does the trick

See **Additional Context** for the logs

### Steps to reproduce

1. Run installer script (to install sshnpd, for example)
2. Kill tmux session that it started
3. Rerun the script/binary
4. Get FileSystemException from ~/.sshnp

### Expected behavior

Do not run into FileSystemException

### Screenshots

_No response_

### Smartphones

_No response_

### Were you using an atApplication when the bug was found?

_No response_

### Additional context

```sh
sshnpd: FileSystemException: lock failed, path = '/home/pi/.sshnp/@server_0/storage/commitLog/commit_log_055719601711c762ce6bc501676e84468ea43ee6a20f32649e09e7cf951c9526.lock' (OS Error: Resource temporarily unavailable, errno = 11)
stack trace: #0      _checkForErrorResponse (dart:io/common.dart:55)
#1      _RandomAccessFile.lock.<anonymous closure> (dart:io/file_impl.dart:1022)
<asynchronous suspension>
#2      StorageBackendVm.initialize (package:hive/src/backend/vm/storage_backend_vm.dart:81)
<asynchronous suspension>
#3      HiveImpl._openBox (package:hive/src/hive_impl.dart:111)
<asynchronous suspension>
#4      HiveImpl.openBox (package:hive/src/hive_impl.dart:142)
<asynchronous suspension>
#5      HiveBase.openBox (package:at_persistence_secondary_server/src/keystore/hive_base.dart:34)
<asynchronous suspension>
#6      CommitLogKeyStore.initialize (package:at_persistence_secondary_server/src/log/commitlog/commit_log_keystore.dart:39)
<asynchronous suspension>
#7      HiveBase.init (package:at_persistence_secondary_server/src/keystore/hive_base.dart:15)
<asynchronous suspension>
#8      AtCommitLogManagerImpl.getCommitLog (package:at_persistence_secondary_server/src/log/commitlog/at_commit_log_manager_impl.dart:26)
<asynchronous suspension>
#9      StorageManager._initStorage (package:at_client/src/manager/storage_manager.dart:26)
<asynchronous suspension>
#10     StorageManager.init (package:at_client/src/manager/storage_manager.dart:14)
<asynchronous suspension>
#11     AtClientImpl._init (package:at_client/src/client/at_client_impl.dart:185)
<asynchronous suspension>
#12     AtClientImpl.create (package:at_client/src/client/at_client_impl.dart:146)
<asynchronous suspension>
#13     DefaultAtServiceFactory.atClient (package:at_client/src/manager/at_client_manager.dart:159)
<asynchronous suspension>
#14     AtClientManager.setCurrentAtSign (package:at_client/src/manager/at_client_manager.dart:80)
<asynchronous suspension>
#15     AtOnboardingServiceImpl._initAtClient (package:at_onboarding_cli/src/onboard/at_onboarding_service_impl.dart:60)
<asynchronous suspension>
#16     AtOnboardingServiceImpl._init (package:at_onboarding_cli/src/onboard/at_onboarding_service_impl.dart:72)
<asynchronous suspension>
#17     AtOnboardingServiceImpl.authenticate (package:at_onboarding_cli/src/onboard/at_onboarding_service_impl.dart:317)
<asynchronous suspension>
#18     _main (file:///sshnoports/bin/sshnpd.dart:146)
<asynchronous suspension>
#19     main (file:///sshnoports/bin/sshnpd.dart:25)
<asynchronous suspension>

Unhandled exception:
FileSystemException: lock failed, path = '/home/pi/.sshnp/@server_0/storage/commitLog/commit_log_055719601711c762ce6bc501676e84468ea43ee6a20f32649e09e7cf951c9526.lock' (OS Error: Resource temporarily unavailable, errno = 11)
#0      _checkForErrorResponse (dart:io/common.dart:55)
#1      _RandomAccessFile.lock.<anonymous closure> (dart:io/file_impl.dart:1022)
<asynchronous suspension>
#2      StorageBackendVm.initialize (package:hive/src/backend/vm/storage_backend_vm.dart:81)
<asynchronous suspension>
#3      HiveImpl._openBox (package:hive/src/hive_impl.dart:111)
<asynchronous suspension>
#4      HiveImpl.openBox (package:hive/src/hive_impl.dart:142)
<asynchronous suspension>
#5      HiveBase.openBox (package:at_persistence_secondary_server/src/keystore/hive_base.dart:34)
<asynchronous suspension>
#6      CommitLogKeyStore.initialize (package:at_persistence_secondary_server/src/log/commitlog/commit_log_keystore.dart:39)
<asynchronous suspension>
#7      HiveBase.init (package:at_persistence_secondary_server/src/keystore/hive_base.dart:15)
<asynchronous suspension>
#8      AtCommitLogManagerImpl.getCommitLog (package:at_persistence_secondary_server/src/log/commitlog/at_commit_log_manager_impl.dart:26)
<asynchronous suspension>
#9      StorageManager._initStorage (package:at_client/src/manager/storage_manager.dart:26)
<asynchronous suspension>
#10     StorageManager.init (package:at_client/src/manager/storage_manager.dart:14)
<asynchronous suspension>
#11     AtClientImpl._init (package:at_client/src/client/at_client_impl.dart:185)
<asynchronous suspension>
#12     AtClientImpl.create (package:at_client/src/client/at_client_impl.dart:146)
<asynchronous suspension>
#13     DefaultAtServiceFactory.atClient (package:at_client/src/manager/at_client_manager.dart:159)
<asynchronous suspension>
#14     AtClientManager.setCurrentAtSign (package:at_client/src/manager/at_client_manager.dart:80)
<asynchronous suspension>
#15     AtOnboardingServiceImpl._initAtClient (package:at_onboarding_cli/src/onboard/at_onboarding_service_impl.dart:60)
<asynchronous suspension>
#16     AtOnboardingServiceImpl._init (package:at_onboarding_cli/src/onboard/at_onboarding_service_impl.dart:72)
<asynchronous suspension>
#17     AtOnboardingServiceImpl.authenticate (package:at_onboarding_cli/src/onboard/at_onboarding_service_impl.dart:317)
<asynchronous suspension>
#18     _main (file:///sshnoports/bin/sshnpd.dart:146)
<asynchronous suspension>
#19     main (file:///sshnoports/bin/sshnpd.dart:25)
<asynchronous suspension>
^C
pi@raspberrypi:~/.local/bin $ rm -rf ~/.sshnp
pi@raspberrypi:~/.local/bin $ ./sshnpd@client_0 
INFO|2023-06-13 21:48:40.937808|AtClientManager|setCurrentAtSign called with atSign @server_0 

INFO|2023-06-13 21:48:40.938400|AtClientManager|Switching atSigns from null to @server_0 

INFO|2023-06-13 21:48:40.946731|HiveBase|commit_log_055719601711c762ce6bc501676e84468ea43ee6a20f32649e09e7cf951c9526 initialized successfully 

INFO|2023-06-13 21:48:40.955055|HiveBase|055719601711c762ce6bc501676e84468ea43ee6a20f32649e09e7cf951c9526 initialized successfully 

INFO|2023-06-13 21:48:40.955562|AtClientCommitLogCompaction (@server_0)|Starting commit log compaction job running for every 11 minute(s) 

INFO|2023-06-13 21:48:40.962861|AtClientManager|setCurrentAtSign complete 

INFO|2023-06-13 21:48:40.963044|AtLookup|Creating new connection 

INFO|2023-06-13 21:48:41.364475|AtLookup|New connection created OK 

INFO|2023-06-13 21:48:41.486333|AtLookup|auth success 

INFO|2023-06-13 21:48:41.692299|AbstractAtKeyEncryption (@server_0)|Encrypted shared symmetric key for @server_0 not found in local storage 

INFO|2023-06-13 21:48:41.692396|AbstractAtKeyEncryption (@server_0)|Deleting @client_0:shared_key@server_0 from LocalSecondary 

INFO|2023-06-13 21:48:41.692930|AbstractAtKeyEncryption (@server_0)|Fetching shared symmetric key for @server_0 from atServer 

INFO|2023-06-13 21:48:41.747406|AbstractAtKeyEncryption (@server_0)|Retrieved my encrypted copy of shared symmetric key for @client_0 from atServer - saving to local storage 

INFO|2023-06-13 21:48:41.759718|AbstractAtKeyEncryption (@server_0)|'Their' copy of shared symmetric key for @client_0 not found in local storage - will check atServer 

INFO|2023-06-13 21:48:41.803355|AbstractAtKeyEncryption (@server_0)|Found 'their' copy of shared symmetric key for @client_0 in atServer - saving to local storage 

INFO|2023-06-13 21:48:42.437406| sshnpd |SUCCESS:id: 6ad23625-0115-45da-93fe-5ed7903a25bf status: NotificationStatusEnum.delivered pi 

WARNING|2023-06-13 21:48:42.453654| sshnpd |connection available 

INFO|2023-06-13 21:48:42.661371|Monitor (@server_0)|monitor started for @server_0 with last notification time: null 

INFO|2023-06-13 21:48:50.070280|AtLookup|Creating new connection 

INFO|2023-06-13 21:48:50.194731|AtLookup|New connection created OK 

INFO|2023-06-13 21:48:50.342376|AtLookup|auth success 

INFO|2023-06-13 21:48:50.386543|SyncService (@server_0)|1071820688|Returning serverCommitId 50 

INFO|2023-06-13 21:48:50.431797|SyncService (@server_0)|1071820688|Returning serverCommitId 50 

SEVERE|2023-06-13 21:48:50.614052|EncryptionUtil|Error while decrypting value: Invalid argument(s): Invalid or corrupted pad block 

WARNING|2023-06-13 21:48:50.614762|SyncService (@server_0)|Exception occurred when setting conflict info for @client_0:shared_key@server_0 | Exception: Invalid argument(s): Invalid or corrupted pad block 

INFO|2023-06-13 21:48:50.676073|SyncService (@server_0)|1071820688|Inside syncComplete. syncRequest.requestSource : SyncRequestSource.system; syncRequest.onDone : Closure: (SyncResult) => void from Function '_onDone@120025363':. 

INFO|2023-06-13 21:48:50.720353|SyncService (@server_0)|1071820688|Returning serverCommitId 54
```

### PR DESCRIPTION
**Pull request by xavierchanth**
**- What I did**

- Made the tar unpacking less verbose so the installer is cleaner
- added logrotation to the non-tmux installation option which closes #175 
- merged updater into installer
  - Checks if the service type is cron, and adds a logrotate line to crontab if it is
  - Checks to see if there was an existing cron entry for this service before trying to start sshnpd (i.e. only start if it is truly the first install)
  closes #203 
- Exits if the tar/zip file is missing
- Escapes "." in ARCH so that the armv7l architecture only matches arm, not arm64

**- How I did it**

**- How to verify it**

**- Description for the changelog**
fix: Installer improvements and bug fixes

### CODE REVIEW

File: scripts/install_sshnpd [hunk 1]
```diff
@@ -13,18 +13,26 @@ norm_atsign() {
...
+  echo "General options:"
+  echo "  -u, --update                           Update all services instead of installing"
   echo "  -l, --local <path>                     Install using local zip/tgz"
   echo "  -r, --repo <path>                      Install using local repo"
   echo "  -h, --help                             Display this help message"
+  echo ""
+  echo "Installation options:"
+  echo "  -c, --client <address>    (mandatory)  Client address (e.g. @alice_client)"
+  echo "  -d, --device <address>    (mandatory)  Device address (e.g. @alice_device)"
+  echo "  -n, --name <device name>  (mandatory)  Name of the device"
```
Comment (JeremyTubongbanua) [inline scripts/install_sshnpd]: Should the mandatory options come first in the usage print?

**Review by JeremyTubongbanua — CHANGES_REQUESTED** [review]

File: scripts/install_sshnpd (hunk 1 above)
Comment (xavierchanth) [reply scripts/install_sshnpd]: They are only mandatory for installation

Comment (xavierchanth) [reply scripts/install_sshnpd]: And technically they aren't mandatory because I ask for them if not provided...

**Comment by xavierchanth** [comment]
@JeremyTubongbanua I've removed the mandatory tag

**Review by JeremyTubongbanua — DISMISSED** [review]
"options" might be confusing for some people maybe, I've seen them called flags

**Comment by xavierchanth** [comment]
From what I've seen it called in the industry:
flags = no value after (--help, -u) are flags
options = value after (-c @‎ client, -d @‎ device, -n devicename) are options

essentially a flag is a boolean option

File: scripts/install_sshnp [hunk 2]
```diff
@@ -157,10 +157,10 @@ parse_env() {
 
   # ARCH includes the dot at the end to avoid conflict between arm and arm64
   case "$(uname -m)" in
-    aarch64|arm64) ARCH="arm64.";;
-    x86_64|amd64) ARCH="x64.";;
-    armv7l|arm) ARCH="arm.";;
-    riscv64) ARCH="riscv64.";;
+    aarch64|arm64) ARCH="arm64\.";;
```
Comment (gkc) [inline scripts/install_sshnp:160]: you could also consider using single quotes

Comment (gkc) [reply scripts/install_sshnp:160]: but if you're later using this in a double-quotes expression, probably best to leave as double quotes :-)

Comment (xavierchanth) [reply scripts/install_sshnp:160]: This is a bug fix that @cconstab found, I want to match a "." at the end of the string using grep, but grep accepts regex. So I need the "\\" in the string to escape the ".".

echo $ARCH outputs "arm64\\." in this case.

The one causing issues was "arm." which was matching both "arm64." and "arm."

Comment (xavierchanth) [reply scripts/install_sshnp:160]: e.g.

```
~/temp 
➜ ls
arm.tgz   arm64.tgz

# with the \

~/temp 
➜ test_ARCH="arm\."   

~/temp 
➜ ls | grep $test_ARCH
arm.tgz

~/temp 
➜ echo $test_ARCH
arm\.

# without the \

~/temp 
➜ test_ARCH="arm." 

~/temp 
➜ ls | grep $test_ARCH
arm.tgz
arm64.tgz

~/temp 
➜ echo $test_ARCH
arm.
```

Comment (xavierchanth) [reply scripts/install_sshnp:160]: > you could also consider using single quotes

See line 173

**Review by cconstab — APPROVED** [review]
tested on armvl pi works fine

**Pull request merged**